File: mobile_gara/lib/models/appointment.dart
class AppointmentModel {
  final String id;
  final int customerId;
  final String appointmentDate;
  final String description;
  final String status; // PENDING, CONFIRMED, CANCELLED
  final String createdAt;

  AppointmentModel({
    required this.id,
    required this.customerId,
    required this.appointmentDate,
    required this.description,
    required this.status,
    required this.createdAt,
  });

  factory AppointmentModel.fromJson(Map<String, dynamic> json) {
    return AppointmentModel(
      id: json['id']?.toString() ?? '',
      customerId: json['customerId'] is int
          ? json['customerId'] as int
          : (int.tryParse(json['customerId']?.toString() ?? '') ?? 0),
      appointmentDate: json['appointmentDate'] ?? '',
      description: json['description'] ?? '',
      status: json['status'] ?? 'PENDING',
      createdAt: json['createdAt'] ?? '',
    );
  }
}

File: backend-services/repair-service/src/main/java/com/gara/repair_service/entity/Appointment.java
package com.gara.repair_service.entity;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "appointments")
@Data
@NoArgsConstructor
public class Appointment {
    
    @Id
    private String id;
    
    @Indexed
    private Long customerId;
    private Long carId;
    
    private LocalDateTime appointmentDate;
    private String description;
    
    // PENDING (Chờ xác nhận), CONFIRMED (Đã xác nhận), CANCELLED (Đã hủy), COMPLETED (Đã hoàn thành)
    @Indexed
    private String status = "PENDING";
    
    private LocalDateTime createdAt = LocalDateTime.now();
}
